svm/src/transaction_processor.rs

```diff
@@ -558,8 +726,19 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         error_counters: &mut TransactionErrorMetrics,
         log_messages_bytes_limit: Option<usize>,
         program_cache_for_tx_batch: &ProgramCacheForTxBatch,
+        unique_loaded_accounts: &mut HashMap<Pubkey, AccountSharedData>,
+        dedup_nonce_lookup: &mut HashSet<Hash>,
     ) -> TransactionExecutionResult {
-        let transaction_accounts = std::mem::take(&mut loaded_transaction.accounts);
+        let mut transaction_accounts = std::mem::take(&mut loaded_transaction.accounts);
+
+        // get latest value of acct struct from unique_loaded_accounts
+        for (key, acct) in transaction_accounts.iter_mut() {
+            if let Some(unique_acct_data) = unique_loaded_accounts.get(key) {
+                if *unique_acct_data != *acct {
```

Comment:
maybe, this incurs byte-wise comparison of Account.data? in that case, that's substantial runtime cost..

Reply:
imo there's no reason to compare here if we take the approach of only loading accounts from the unique_loaded_accounts here and not in the initial fn call. See my previous comment on this.